Properly Placed Pages: new page cell

danielebogo opened the pull request:
This PR updates the table view cell for the Page list, adding the featured image and the time stamp. Also the badges are replaced by a normal label.

![simulator screen shot - iphone x - 2019-01-14 at 17 21 21](https://user-images.githubusercontent.com/912252/51131266-28465580-1827-11e9-962b-a13a00b37984.png)

**To Test:**
- Select a blog and open the pages list
- Set a page as Private and/or Review Required to see the badge label
- Set a feature image to see it on the pages list

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

iamthomasbishop commented:
Looking good, @danielebogo! One question: when there is a thumbnail, what happens when the text of the date + status is long? Does the text break to the next line?

// cc @SylvesterWilmott for alignment – does this align with the Post card/list designs you worked on last week?

aerych commented on WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m:
```diff
@@ -114,14 +140,41 @@ - (void)configureBadges
-    if (page.hasPendingReviewState) {
-       self.privateBadgeLabel.text = NSLocalizedString(@"Pending review", @"Title of the Pending Review Badge");
+    NSString *badgesString = @"";
+    BOOL displayFirstBadge = page.hasPrivateState || page.hasPendingReviewState;
```
I wonder if the logic in this method could be simplified.  
Currently its one conditional check to assign the value of displayFirstBadge, then another check to see if we need to assign a string value, then another check to determine if the dash spacer.
Would it be nicer to do something like this? 
``` 
if (page.hasPendingReviewState) {
  // assign string
} else if (page.hasPendingReviewState) {
  // assign string
}

if (page.hasLocalChanges) {
  if ( badgeString.length > 0) {
    // append string
  }
  // append string
}
```
It would avoid the need to have `displayFirstBadge`.   Thoughts?

danielebogo replied:
Yep! Much better!

SylvesterWilmott commented:
@iamthomasbishop 

>One question: when there is a thumbnail, what happens when the text of the date + status is long? Does the text break to the next line?

It's truncated to maintain vertical rhythm but I'd be interested in adding the statuses within the page/post edit view.

>does this align with the Post card/list designs you worked on last week?

Yes it does. We'll recycle these list styles for the post list for sure.

danielebogo commented on WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m:
```diff
@@ -78,23 +107,20 @@ - (void)applyStyles
     
     self.titleLabel.textColor = [WPStyleGuide darkGrey];
+    self.timestampLabel.textColor = [WPStyleGuide grey];
+    self.badgesLabel.textColor = [UIColor UIColorFromHex:0xd89511];
```
Yep I agree. @SylvesterWilmott do you have a name for this and also other colors to add?

SylvesterWilmott replied:
This is kind of like a stop-gap solution to our yellow being really super unreadable when applied to text. Would it be possible to tackle this when we add the new colour palette?

danielebogo replied:
@aerych @SylvesterWilmott I can create a _temporary_ extension for WP iOS that will be replaced by the new colour palette in future. What do you think?

aerych replied:
This sounds good.  I think an extension is less likely to get over looked by a future iteration. Good compromise 👍

danielebogo commented:
Thanks everyone to review this!
@aerych I fixed everything and added the extension
@SylvesterWilmott @iamthomasbishop the font for the set parent view has been changed to system font size 17

aerych commented:
Marking approved, but there is a last minute question of whether the intent is to show an image from the page, or if the intent is to show a page's actual featured image.  If we're just trying to show an image from the page in the thumbnail then this is good to go (so no need for me to be a bottleneck).  On the other hand, if we want to only ever show the featured image then we can make those changes and review again.

danielebogo commented:
@aerych thanks for that! The intent is to show only the featured image. I fixed it using the _Media_ property _featuredImage_. Also the image loader accept directly a Media!

danielebogo commented:
@SylvesterWilmott I fixed it!

![simulator screen shot - iphone x - 2019-01-16 at 12 59 08](https://user-images.githubusercontent.com/912252/51248615-fa7d2000-1990-11e9-839b-a1de02015385.png)

SylvesterWilmott commented:
Looks great! thanks @danielebogo

danielebogo commented:
@iamthomasbishop good catch! I changed the image size requested by the image loader.

aerych commented:
Looks good!  :shipit:

danielebogo commented:
Thanks everyone!
